agent_pool.go

```diff
@@ -0,0 +1,190 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ AgentPools = (*agentPools)(nil)
+
+// AgentPools describes all the agent pool related methods that the Terraform
+// Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/agents.html
+type AgentPools interface {
+	// List all the agent pools of the given organization.
+	List(ctx context.Context, organization string, options AgentPoolListOptions) (*AgentPoolList, error)
+
+	// Create a new agent pool with the given options.
+	Create(ctx context.Context, organization string, options AgentPoolCreateOptions) (*AgentPool, error)
+
+	// Read a agent pool by its ID.
+	Read(ctx context.Context, agentPoolID string) (*AgentPool, error)
+
+	// Update an agent pool by its ID.
+	Update(ctx context.Context, agentPool string, options AgentPoolUpdateOptions) (*AgentPool, error)
+
+	// Delete an agent pool by its ID.
+	Delete(ctx context.Context, agentPoolID string) error
+}
+
+// agentPools implements AgentPools.
+type agentPools struct {
+	client *Client
+}
+
+// AgentPoolList represents a list of agent pools.
+type AgentPoolList struct {
+	*Pagination
+	Items []*AgentPool
+}
+
+// AgentPool represents a Terraform Enterprise agent pool.
+type AgentPool struct {
+	ID   string `jsonapi:"primary,agent-pools"`
+	Name string `jsonapi:"attr,name"`
+}
+
+// AgentPoolListOptions represents the options for listing agent pools.
+type AgentPoolListOptions struct {
+	ListOptions
+}
+
+// List all the agent pools of the given organization. Note this currently is
+// limited to a single pool per organization (API enforced).
```

Comment:
nit: remove the part about single agent pool
```suggestion
// List all the agent pools of the given organization.
```